ui/pages/confirmations/components/confirm/info/hooks/known-method-data-in-transaction.ts

```diff
@@ -0,0 +1,22 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  getKnownMethodData,
+  use4ByteResolutionSelector,
+} from '../../../../../../selectors';
+import { getContractMethodData } from '../../../../../../store/actions';
+
+export const useKnownMethodDataInTransaction = (
+  currentConfirmation: TransactionMeta,
+) => {
+  const dispatch = useDispatch();
+  const use4ByteResolution = useSelector(use4ByteResolutionSelector);
+  if (use4ByteResolution && currentConfirmation?.txParams?.data) {
```

Comment:
Minor, but could we define a variable such as `transactionData` so we don't need to reference the full chain multiple times?